Patch the locales from flutter, with the correct runtime information (decimal seperator only for now)

BentEngbers opened the pull request:
I have written a simple wrapper, that will enable users of this package to patch the decimal seperator of the flutter locales in `numberFormatSymbols`. This wil fix issues in apps bases that depend on these locales having the correct decimal separator, which is not always the case. Instead of having two sources of decimal seperators (locale & locale_plus), and putting async calls in the build method, the developer can use the normal locales as the correct source.  It works on all platforms except web.
- [x] This functionality could be expanded for all functions all calls in the `LocalePlus` class.
## Pre-launch Checklist

- [x] I updated `pubspec.yaml` with an appropriate new version.
- [x] I updated `CHANGELOG.md` with new version number and description.
- [x] I updated `README.md`.
- [x] I updated example project with new feature.
- [x] Check if the branch name is correct.
- [x] Code is warning free.

gokberkbar commented:
Hello @BentEngbers , thank you for your contribution. Right now `locale_plus` can only override `DECIMAL_SEP` & `GROUP_SEP` from `NumberSymbols`. I think this feature can override both of them in a single function. Doing it separately can cause bugs. Can you also implement overriding grouping separator in single function and update function names `README.md` etc. accordingly. Right now `locale_plus`  only support android and iOS platforms so web platform based problems is not an issue. After that I will be happy to merge your pull request and release new version.

BentEngbers commented:
Hi @gokberkbar! 
I have added `GROUP_SEP` patching.
I also use the [universal io](https://pub.dev/packages/universal_io) package to also enable the `patchNumberSeperators` function on web. I have updated the readme and the example file.
